mlflow/pipelines/steps/split.py

```diff
@@ -61,10 +61,34 @@ def _get_split_df(input_df, hash_buckets, split_ratios):
     return train_df, validation_df, test_df
 
 
+def _parallelize(data, func, num_of_processes=8):
+    import numpy as np
+    import pandas as pd
+    from multiprocessing import Pool
+
+    data_split = np.array_split(data, num_of_processes)
+    pool = Pool(num_of_processes)
+    data = pd.concat(pool.map(func, data_split))
+    pool.close()
+    pool.join()
+    return data
+
+
+def _run_on_subset(func, data_subset):
+    return data_subset.applymap(func)
```

Comment:
Can probably move this into `_parallelize_on_rows` as local function

Reply:
This creates some issues with pickle -- see this [stackoverflow thread](https://stackoverflow.com/questions/52265120/python-multiprocessing-pool-attributeerror).